### DISCUSSION TITLE
Make monitoring of  healthchecks.io easier by instrumenting metrics / traces.

### DISCUSSION THREAD

**Discussion by jmichalek132** [healthchecks/healthchecks#529] [General]
Hi, I would like to instrument healthchecks.io with proper metrics (i.e. tracking http requests duration, status codes, etc...). So we monitoring of the app in production is a lot easier. Ideally using which supports also traces. https://opentelemetry.io/docs/python/

**Comment by cuu508** [comment]
A couple thoughts / pointers:

* You can monitor request duration and status codes from the outside (e.g. by analyzing web server's access logs). You may not need to instrument python code for this
* There are or could be internal metrics like "alert dwell time" (time from alert being generated to it actually being sent out)  that would need instrumentation. I've used statsd for this – take a look at [sendalerts.py](https://github.com/healthchecks/healthchecks/blob/master/hc/api/management/commands/sendalerts.py)
* for exception logging I'm using Sentry

**Reply by jmichalek132** [reply]
We prefer to use native metrics when possible, so we don't need to depend add proxy we don't need, and extra steps for getting metrics out of it.
